src/Processors/Transforms/buildPushingToViewsChain.cpp

```diff
@@ -118,7 +118,7 @@ class ExecutingInnerQueryFromViewTransform final : public ExceptionKeepingTransf
     struct State
     {
         QueryPipeline pipeline;
-        PullingPipelineExecutor executor;
```

Comment:
I'm afraid this can be too much, since usually the block is pretty small, so for instance in your example it is only 9% faster, while it uses 8 threads. And I think that there can be lots of examples where it will be even slower.

Of course it should be more beneficial in case of bigger blocks, however this is not a usual thing.

In my opinion this does not worth it, however if others will thing differently, at least it should be done under some setting to avoid possible degradation for other patterns.